src/sickle.ts

```diff
@@ -159,32 +223,83 @@ class Annotator {
   }
 
   private emitFunctionType(fnDecl: ts.FunctionLikeDeclaration) {
-    // The first \n makes the output sometimes uglier than necessary,
-    // but it's needed to work around
-    // https://github.com/Microsoft/TypeScript/issues/6982
-    this.emit('\n/**\n');
-    let existingAnnotation = this.existingClosureAnnotation(fnDecl);
-    if (existingAnnotation) {
-      this.emit(' * ' + existingAnnotation + '\n');
+    // Construct the JSDoc comment by reading the existing JSDoc, if
+    // any, and merging it with the known types of the function
+    // parameters and return type.
+    let jsDoc = this.getJSDoc(fnDecl);
+    let newDoc: JSDocComment = {tags: []};
```

Comment:
I need to shuffle together @params from two sources: there are some (but not necessarily all) in the existing jsdoc, and the source of truth comes from TypeScript type info.

If I wanted to only augment the existing jsdoc I'd need to find all the @param tags in it, if any, then find the right places to insert the params known by TypeScript amongst them.  I think it's simpler code to create a new data structure.